src/Network/Network/VirtualNetworkGateway/NewAzureVirtualNetworkGatewayCommand.cs

```diff
@@ -260,6 +264,16 @@ public class NewAzureVirtualNetworkGatewayCommand : VirtualNetworkGatewayBaseCmd
         [ValidateNotNullOrEmpty]
         public string[] CustomRoute { get; set; }
 
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The generation for this VirtualNetwork VPN gateway. Must be None if GatewayType is not VPN.")]
+        [PSArgumentCompleter(
+            MNM.VpnGatewayGeneration.None,
+            MNM.VpnGatewayGeneration.Generation1,
+            MNM.VpnGatewayGeneration.Generation2)]
+        [ValidateNotNullOrEmpty]
```

Comment:
use `[ValidateSet("None", "Generation1", "Generation2" , IgnoreCase = true)]` instead of `ValidateNotNullOrEmpty`